docs/source/index.rst

```diff
@@ -54,52 +54,33 @@ easy.
 **Step 1:** Pick the magic string that will identify your library. To
 avoid collisions, this should match your library's name on PyPI.
 
-**Step 2:** There's a special :class:`contextvars.ContextVar` object:
+**Step 2:** There's a special :class:`threading.local` object:
 
-.. data:: current_async_library_cvar
+.. data:: current_async_library_tlocal.name
 
-Make sure that whenever your library is running, this is set to your
-identifier string. In most cases, this will be as simple as:
+Make sure that whenever your library is calling a coroutine throw(), send(), or close()
+that this is set to your identifier string. In most cases, this will be as simple as:
 
 .. code-block:: python3
 
-   from sniffio import current_async_library_cvar
+   from sniffio import current_async_library_tlocal
 
-   # Your library's run function
-   def run(...):
-        token = current_async_library_cvar.set("my-library's-PyPI-name")
+   # Your library's step function
+   def step(...):
+        old_name, current_async_library_tlocal.name = (
+            current_async_library_tlocal.name,
+            "my-library's-PyPI-name",
+         )
         try:
-            # The actual body of your run() function:
-            ...
+            result = coro.send(None)
         finally:
-            current_async_library_cvar.reset(token)
+            current_async_library_tlocal.name = old_name
 
 **Step 3:** Send us a PR to add your library to the list of supported
 libraries above.
 
 That's it!
 
-Notes:
-
-On older Pythons without native contextvars support, sniffio
-transparently uses `the official contextvars backport
-<https://pypi.org/project/contextvars/>`__, so you don't need to worry
-about that.
-
-There are libraries that can switch back and forth between different
-async modes within a single call-task – like ``trio_asyncio`` or
-Twisted's asyncio operability. These libraries should make sure to set
-the value back and forth at appropriate points.
-
-The general rule of thumb: :data:`current_async_library_cvar` should
-be set to X exactly at those moments when ``await X.sleep(...)`` will
-work.
-
-.. warning:: You shouldn't attempt to read the value of
-   ``current_async_library_cvar`` directly –
-   :func:`current_async_library` has a little bit more cleverness than
-   that.
```

Comment:
Hmm, yeah, this text all seems like it's still true, just with the threadlocal instead of the cvar.

Reply:
I think these warnings no-longer apply - you just make the switch before calling `throw(), send(), or close()` on a coroutine.

Reply:
> There are libraries that can switch back and forth between different
async modes within a single call-task – like ``trio_asyncio`` or
Twisted's asyncio operability. 

eg this doesn't apply to twisted because an `asyncio.Handle` callback can't be called from a `generator.send`

Reply:
Isn't it just a reminder to think about the high level cases while working down in the low level implementation?  But sure, there's what, three libraries that implement this...  But I guess this discussion would continue in the suggested change below, if at all.

Reply:
in the current state of the PR we're advising libraries to read the the value of ``current_async_library_tlocal`` directly. And `current_async_library_cvar` is now undocumented